src/snowflake/snowpark/mock/_connection.py

```diff
@@ -322,9 +342,18 @@ def run_query(
         ] = None,  # this argument is currently only used by AsyncJob
         **kwargs,
     ) -> Union[Dict[str, Any], AsyncJob]:
-        raise NotImplementedError(
-            "[Local Testing] Running SQL queries is not supported."
-        )
+        pattern = r"^use (warehouse|database|schema|role) .+$"
+        if re.match(pattern, query):
+            first_space_index = query.find(" ")
+            second_space_index = query.find(" ", first_space_index + 1)
+            object_type = query[first_space_index + 1 : second_space_index]
+            object_name = query[second_space_index + 1 :]
+            setattr(self, f"_active_{object_type}", object_name)
+            return {"data": [("Statement executed successfully.",)], "sfqid": None}
```

Comment:
I think this could be a little streamlined.
```suggestion
        pattern = r"^\s*use\s+(warehouse|database|schema|role)\s+(\w+)\s*$"
        if match := re.match(pattern, query):
            object_type = match.group(1)
            object_name = match.group(2)
            setattr(self, f"_active_{object_type}", object_name)
            return {"data": [("Statement executed successfully.",)], "sfqid": None}
```
This is assuming object names are made up of word characters.

Reply:
thanks for the suggestion! I adopted most of your proposal, however, object name can contain special chars, and when the code execution arrives here, the object name is already validated in the caller.